api-service/webapp-demo/src/app/test_job.component.ts

```diff
@@ -62,191 +42,29 @@ export class TestJobComponent implements OnInit {
   hateAnswer: string;
   comments: string;
 
-  loading: boolean;
-
-  training_answer_count: number = 0;
-  user_mean_score: number = 0;
-
-  overall_job_answer_count: number = 0;
-  overall_job_mean_score: number = 0;
-
-  // Saved into local storage and used to measure the number of sent
-  // requests from the browser's prespective.
-  local_sent_count: number = 0;
-
-  constructor(private router: Router,
-              //private params: RouteParams,
-              private route: ActivatedRoute,
-              private crowdSourceApiService: CrowdSourceApiService) {}
-
-  ngOnInit(): void {
-    // Determine if page should render in embedded mode
-    var query = document.location.search.substr(1)
-    var embedded_query = /embedded=(true|false)/g.exec(query);
-    if (embedded_query) {
-      this.notEmbedded = !(embedded_query[1] == 'true');
-    }
-
-    this.userNonce = localStorage.getItem('user_nonce');
-    const maybe_local_sent_count = localStorage.getItem('local_sent_count');
-    if (maybe_local_sent_count !== null) {
-      this.local_sent_count = parseInt(maybe_local_sent_count, 10);
-    } else {
-      this.local_sent_count = 0;
-      localStorage.setItem(
-          'local_sent_count', this.local_sent_count.toString());
-    }
-
-    if (!this.userNonce) {
-      this.userNonce = Math.random().toString();
-      localStorage.setItem('user_nonce', this.userNonce);
-    }
-    console.log(`user_nonce: ` + this.userNonce);
-
-    // TODO(rachelrosen): Refactor this code to use a more asynchronous pattern,
-    // e.g. nextWork$ = this.route.paramMap.switchMap((params: ParamMap) =>{...});
-    this.route.paramMap.subscribe((params: ParamMap) => {
-      this.updateIds(params);
-    });
-
-    this.getNextWorkItem();
-  }
-
-  updateIds(params: ParamMap): void {
-    this.customClientJobKey = params.get('customClientJobKey');
-    this.questionId = params.get('questionId');
+  public buildAnswer(): TestJobCrowdSourceAnswer {
+    return Object.assign(
+      super.buildAnswer(),
+      {
+        questionId: this.questionId,
```

Comment:
Just to check my understanding, we have this `questionId` field for clarity, because the base class will fill in this field otherwise. Is that right?